libyul/optimiser/UnusedStoreEliminator.cpp

```diff
@@ -78,17 +78,17 @@ void UnusedStoreEliminator::run(OptimiserStepContext& _context, Block& _ast)
 		ignoreMemory
 	};
 	rse(_ast);
-	if (
-		auto evmDialect = dynamic_cast<EVMDialect const*>(&_context.dialect);
-		evmDialect && evmDialect->providesObjectAccess()
-	)
-		rse.changeUndecidedTo(State::Unused, Location::Memory);
+
+	auto evmDialect = dynamic_cast<EVMDialect const*>(&_context.dialect);
+	if (evmDialect && evmDialect->providesObjectAccess())
+		rse.clearActive(Location::Memory);
 	else
-		rse.changeUndecidedTo(State::Used, Location::Memory);
-	rse.changeUndecidedTo(State::Used, Location::Storage);
-	rse.scheduleUnusedForDeletion();
+		rse.markActiveAsUsed(Location::Memory);
+	rse.markActiveAsUsed(Location::Storage);
+	rse.m_storesToRemove += rse.m_allStores - rse.m_usedStores;
 
-	StatementRemover remover(rse.m_pendingRemovals);
+	set<Statement const*> toRemove{rse.m_storesToRemove.begin(), rse.m_storesToRemove.end()};
```

Comment:
Is this copy necessary - i.e. could you pass `rse.m_storesToRemove` directly here?

Reply:
Unfortunately yes (at least in its current form) - `StatementRemover` accepts a const reference to a `map`, and `m_storesToRemove` is a `vector`, so we this conversion needs to happen. In addition, the `m_toRemove` members is also a const reference in `StatementRemover`, so can't move without a refactor either. I'll try doing a small refactor to see if this yields an improvement - probably won't be worth it in the end, because the approach in this PR already cut the runtime (or compile time) significantly.